databroker/experimental/server_ext.py

```diff
@@ -125,6 +127,54 @@ def put_data(self, body):
         if result.matched_count != result.modified_count:
             raise ValueError("Error while writing to database")
 
+    def put_metadata(self, metadata, specs):
+
+        revisions_doc = (
+            self.revisions.find({"key": self.doc.key}).sort({"revision": -1}).limit(1)
+        )
+        if revisions_doc is not None:
```

Comment:
We should look at ways to do an atomic increment in MongoDB. There is a race condition here if two clients do updates concurrently. We need to ensure the `revision` is unique.

Reply:
Note: we're rely on an index. There is a chance of a race, but the result will be failure (and client can try again...) rather than duplication.